lib/master_lock.rb

```diff
@@ -56,6 +56,8 @@ class << self
     # @raise [LockNotAcquiredError] if the lock cannot be acquired before the
     #   timeout
     def synchronize(key, options = {})
+      lock_acquired_start_time = Time.now # used for benchmarking
```

Comment:
The lock has not been acquired yet. I don't think you are measuring the right thing.